volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -236,53 +264,53 @@ def parse_data_record(
             content = renderers.NotAvailableValue()
 
         yield (
-            format_hints.Hex(record_map[mft_record.vol.offset][2]),
-            mft_record.get_signature(),
+            format_hints.Hex(record_map[mft_record.vol.offset].offset),
+            str(mft_record.get_signature()),
             mft_record.RecordNumber,
             attr.Attr_Header.AttrType.lookup(),
-            record_map[mft_record.vol.offset][0],
+            record_map[mft_record.vol.offset].record_name
+            or renderers.NotAvailableValue(),
             ads_name,
             content,
         )
 
     @classmethod
     def parse_data_records(
         cls,
-        record_map: Dict[int, Tuple[str, int, int]],
-        mft_record: interfaces.objects.ObjectInterface,
-        attr: interfaces.objects.ObjectInterface,
+        record_map: DefaultDict[str, MFTRecord],
```

Comment:
```suggestion
        record_map: Dict[str, MFTRecord],
```
Again, pretty sure type can just be `Dict[str, MFTRecord]`